dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java

```diff
@@ -612,6 +612,7 @@ public void addItem(Context context, Collection collection, Item item) throws SQ
         // Instead we add the collection to an item which works in the same way.
         if (!item.getCollections().contains(collection)) {
             item.addCollection(collection);
+            item.setModified();
```

Comment:
I would prefer to modify the IndexConsumer to deal properly with the ADD event in a collection instead than flag the item is updated when it is indeed potentially unchanged. This avoid the need to reprocess the item by other consumer that are not really interested in the collection mapping and could lead to more efficient implementation